code/app/src/main/java/com/example/bread/repository/ParticipantRepository.java

```diff
@@ -115,4 +163,73 @@ public void checkIfAlreadyFollowed(@NonNull String username, String followerUser
 
     public void checkIfAlreadyFollowing(@NonNull String username, String followingUsername, @NonNull OnSuccessListener<Boolean> onSuccessListener, OnFailureListener onFailureListener) {
     }
+
+    public void acceptFollowRequest(String requestId, String senderId, String receiverId, OnSuccessListener<Void> onSuccessListener, OnFailureListener onFailureListener) {
+        WriteBatch batch = db.batch();
+
+        DocumentReference senderRef = db.collection("participants").document(senderId);
+        DocumentReference receiverRef = db.collection("participants").document(receiverId);
+        DocumentReference requestRef = db.collection("followRequests").document(requestId);
+
+        // Delete the follow request
+        batch.delete(requestRef);
+
+        // Conditionally add to following and followers list if not already added
+        senderRef.get().addOnSuccessListener(senderSnapshot -> {
+            List<String> following = (List<String>) senderSnapshot.get("following");
+            if (following == null || !following.contains(receiverId)) {
+                batch.update(senderRef, "following", FieldValue.arrayUnion(receiverId));
```

Comment:
Why aren’t we using the provided methods? Also, I originally envisioned followers and following as collections rather than arrays, so arrayUnion wouldn’t work as expected. If you’d like, we can discuss my approach and how I imagine this functionality should work.